src/service/programService.js

```diff
@@ -219,6 +219,74 @@ async function deleteProgram(req, response) {
   });
 }
 
+function getProgramCountsByOrg(req, response) {
+  var data = req.body
+  var rspObj = req.rspObj
+
+  rspObj.errCode = programMessages.PROGRAMCOUNTS_BYORG.PROGRAMCOUNTS_FETCH.FAILED_CODE
+  rspObj.errMsg = programMessages.PROGRAMCOUNTS_BYORG.PROGRAMCOUNTS_FETCH.FAILED_MESSAGE
+  rspObj.responseCode = '';
+
+  const findQuery = (data.request && data.request.filters) ? data.request.filters : {}
+  const facets = ["rootorg_id"];
+  model.program.findAll({
+    where: {
+      ...findQuery
+    },
+    attributes: [...facets, [Sequelize.fn('count', Sequelize.col(facets[0])), 'count']],
+    group: [...facets]
+  }).then((result) => {
+      const apiRes = _.keyBy(_.map(result, 'dataValues'), 'rootorg_id');
+
+      const orgIds = _.map(apiRes, 'rootorg_id');
+      if (_.isEmpty(result) || _.isEmpty(orgIds)) {
+        return response.status(200).send(successResponse(rspObj));
+      }
+
+      getOrganisationDetails(req, orgIds).then((orgData) => {
+        _.forEach(orgData.data.result.response.content, function(el, index){
+          el.program_count = apiRes[el.id].count;
+        });
+        rspObj.result = orgData.data.result.response;
+        return response.status(200).send(successResponse(rspObj));
+      }, (error) => {
+        rspObj.responseCode = responseCode.SERVER_ERROR
+        rspObj.errCode = programMessages.PROGRAMCOUNTS_BYORG.ORGSEARCH.FAILED_CODE
+        rspObj.errMsg = programMessages.PROGRAMCOUNTS_BYORG.ORGSEARCH.FAILED_MESSAGE
+        loggerError(rspObj.errMsg, rspObj.errCode, rspObj.errMsg, error.response.data.responseCode, data, req)
+        rspObj.result = err;
+        return response.status(400).send(errorResponse(rspObj));
+      })
+  }).catch((err) => {
+    rspObj.responseCode = responseCode.SERVER_ERROR
+    loggerError('Error fetching program count group by facets',
+    rspObj.errCode, rspObj.errMsg, rspObj.responseCode, err, req);
+    rspObj.result = err;
```

Comment:
Don't pass the DB err back in the response. it might give away the auth tokens.

Reply:
removed it, The rspObj.result is bydefault set to {}. We are setting errCode and errMsg alredy.